### PR TITLE
Todo-Management-Project Progress

### PR DESCRIPTION
**Pull request by Aman-8273**
todo-project progress

### CODE REVIEW

File: src/components/TodoList.tsx [hunk 1]
```diff
@@ -13,8 +13,12 @@ import {
 import { TodoListProps } from '../types';
 import EditRoundedIcon from '@mui/icons-material/EditRounded';
 import DeleteRoundedIcon from '@mui/icons-material/DeleteRounded';
+import { useMutation, useQuery } from '@apollo/client';
+import { CREATE_OR_UPDATE_TODO, GET_TODOS } from '../graphql/todoQuries';
 
 const TodoList = ({ items, onDeleteTodos, handleEdit }: TodoListProps) => {
+  console.log(items);
```
Comment (bhasker29) [inline src/components/TodoList.tsx]: use console debug always

File: src/components/TodoList.tsx [hunk 2]
```diff
@@ -23,34 +27,62 @@ const TodoList = ({ items, onDeleteTodos, handleEdit }: TodoListProps) => {
...
+
+  const [updateStatus] = useMutation(CREATE_OR_UPDATE_TODO);
+
+  const handleStatusChange = async (
+    id: string,
+    currentStatus: string,
+    title: string,
+    description: string
+  ) => {
+    const newStatus = currentStatus === 'PENDING' ? 'COMPLETED' : 'PENDING';
```
Comment (bhasker29) [inline src/components/TodoList.tsx]: 'PENDING' ? 'COMPLETED' : 'PENDING';

Use constants

File: src/components/TodoList.tsx [hunk 3]
```diff
@@ -23,34 +27,62 @@ const TodoList = ({ items, onDeleteTodos, handleEdit }: TodoListProps) => {
...
+
+    try {
+      await updateStatus({
+        variables: {
+          input: {
+            id,
+            status: newStatus,
+            email: 'krips@mail.com',
+            title,
```
Comment (bhasker29) [inline src/components/TodoList.tsx:69]: Take this value from localstorage
            email: 'krips@mail.com',

File: src/components/TodoPage.tsx [hunk 4]
```diff
@@ -14,51 +31,100 @@ const TodoPage = () => {
...
+    //     : todo
+    // );
+    // setAddTodo(updatedTask);
+
+    //TODO
+    await createOrUpdateTodo({
+      variables: {
+        input: {
+          id,
+          email: 'krips@mail.com',
```
Comment (bhasker29) [inline src/components/TodoPage.tsx]: email: 'krips@mail.com',
   take this from localstorage

File: src/main.tsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ const root = ReactDOM.createRoot(
 );
 
 const client = new ApolloClient({
-  uri: 'https://graphqlapi.com/',
+  uri: 'https://b321-2409-40c1-5023-badb-a914-fd50-5db2-d65c.ngrok-free.app/graphql',
```
Comment (bhasker29) [inline src/main.tsx]: take this form the configurations

**Review by bhasker29 — CHANGES_REQUESTED** [review]

**Comment by Aman-8273** [comment]
Ok sir, I do it tomorrow..

**Comment by Aman-8273** [comment]
Screenshot (44)

**Pull request merged**